app/common/alerts.py

```diff
@@ -14,12 +14,25 @@ class AlertViaGovNotify:
 
     @staticmethod
     def send(email, reference):
-        notifications_client = NotificationsAPIClient(settings.NOTIFICATION_COMBINED_KEY)
-        notifications_client.send_email_notification(email_address=email,
-                                                     template_id=settings.NOTIFICATION_TEMPLATE_ID,
-                                                     personalisation=None,
-                                                     reference=reference)
-        logger.info('Sent secure message email notification', user_uuid=g.user.user_uuid)
+
+        try:
+            notification = {
+                "emailAddress": email,
+                "reference": reference
+            }
+
+            url = urlparse.urljoin(settings.RM_NOTIFY_GATEWAY_URL, settings.NOTIFICATION_TEMPLATE_ID)
+
+            response = requests.post(url, auth=settings.BASIC_AUTH, timeout=settings.REQUESTS_POST_TIMEOUT,
+                                     json=notification)
+
+            logger.info('Sent secure message email notification, via RM Notify-Gateway to GOV.UK Notify.',
+                        message_id=response.json()["id"])
+
+        except Exception as e:
+            msg = 'There was a problem sending a notification via RM Notify-Gateway to GOV.UK Notify'
+            logger.error(msg, exception=e)
+            raise RasNotifyException(msg + str(e), status_code=500)
```

Comment:
So now it will return 500 which means in back-stage it will show 500 error page even though the secure message has been 'sent', instead of failing quietly like it did before. Is this correct or do we need to raise ticket for backstage to handle this scenario differently?

Reply:
We should be catching the explicit exceptions which could be thrown and handle them correctly. If the message is sent we shouldn't fall over. What exception are we handling here?

Does `msg + str(e)` give anything useful in the message?

Reply:
I previously said that it should continue normally? The code was logging the error but continuing.